lib/puppet/pops/types/type_formatter.rb

```diff
@@ -290,18 +290,26 @@ def to_s
   end
 
   NAME_SEGMENT_SEPARATOR = '::'.freeze
+  STARTS_WITH_ASCII_CAPITAL = /^[A-Z]/
 
-  # Capitalizes each segment in a name separated with the {NAME_SEPARATOR}
+  # Capitalizes each segment in a name separated with the {NAME_SEPARATOR} conditionally. The name
+  # will not be subject to capitalization if it already starts with a capital letter. This to avoid
+  # that existing camel casing is lost.
+  #
   # @param qualified_name [String] the name to capitalize
   # @return [String] the capitalized name
   #
   def capitalize_segments(qualified_name)
-    segments = qualified_name.split(NAME_SEGMENT_SEPARATOR)
-    if segments.size == 1
-      qualified_name.capitalize
+    if !qualified_name.is_a?(String) || qualified_name =~ STARTS_WITH_ASCII_CAPITAL
```

Comment:
It is an API public method, it breaks the API as it will potentially not correctly capitalize a string if it starts with an initial upper case letter. Sorry.